test/lib/pipeline.test.js

```diff
@@ -299,6 +305,32 @@ describe('Pipeline Model', () => {
         });
     });
 
+    describe('get secrets', () => {
+        it('has a secrets getter', () => {
+            const listConfig = {
+                params: {
+                    pipelineId: pipeline.id
+                },
+                paginate: {
+                    count: 25, // This limit is set by the matrix restriction
+                    page: 1
+                }
+            };
+
+            secretFactoryMock.list.resolves(null);
+            // when we fetch secrets it resolves to a promise
+            assert.isFunction(pipeline.secrets.then);
+            // and a factory is called to create that promise
+            assert.calledWith(secretFactoryMock.list, listConfig);
+
+            // When we call pipeline.secrets again it is still a promise
+            assert.isFunction(pipeline.secrets.then);
+            // ...but the factory was not recreated, since the promise is stored
+            // as the model's pipeline property, now
```

Comment:
It's a getter so it's a function. But you access it using `pipeline.secrets`

The test for this getter is written the same way as how we wrote it for other getters. For example here: https://github.com/screwdriver-cd/models/blob/6d31261ea12a0fe2df2ac721d9a074399ba64fc0/test/lib/pipeline.test.js#L260-L274

If we want to change it, we should do it in another PR where we change all of them. But imo, it tests the same thing, just different way of writing it. You can ask @petey why it was written in the previous way, maybe he has better reasons.